src/v3/src/transformer/terminal/transformTerminalTransaction.ts

```diff
@@ -119,6 +119,8 @@ const appendViewLinks = (
       cancelLink.options.href = backToSigninUri;
     } else if (isOauth2Enabled(widgetProps)) {
       cancelLink.options.onClick = async () => {
+        // We have to directly delete the recoveryToken since it is set once upon authClient instantiation
+        delete authClient?.options.recoveryToken;
```

Comment:
If you add the clear in the `useOnSubmit` hook (within the condition), can you also add that here?